src/axom/quest/examples/quest_candidates_example.cpp

```diff
@@ -0,0 +1,754 @@
+    : axom::execution_space<ExecSpace>::allocatorID();
+
+  // Copy the insert-BVH hexes to the device, if necessary
+  // Either way, insert_hexes_v will be a view w/ data in the correct space
+  auto& insert_hexes_h = insertMesh.hexes();
+  HexArray insert_hexes_d =
+    on_device ? HexArray(insert_hexes_h, kernel_allocator) : HexArray();
+
+  // Copy the insert-BVH bboxes to the device, if necessary
+  // Either way, insert_bbox_v will be a view w/ data in the correct space
+  auto& insert_bbox_h = insertMesh.hexBoundingBoxes();
+  BBoxArray insert_bbox_d =
+    on_device ? BBoxArray(insert_bbox_h, kernel_allocator) : BBoxArray();
+  auto insert_bbox_v = on_device ? insert_bbox_d.view() : insert_bbox_h.view();
+
+  // Copy the query-BVH hexes to the device, if necessary
+  // Either way, query_hexes_v will be a view w/ data in the correct space
+  auto& query_hexes_h = queryMesh.hexes();
+  HexArray query_hexes_d =
+    on_device ? HexArray(query_hexes_h, kernel_allocator) : HexArray();
+
+  // Copy the query-BVH bboxes to the device, if necessary
+  // Either way, bbox_v will be a view w/ data in the correct space
+  auto& query_bbox_h = queryMesh.hexBoundingBoxes();
+  BBoxArray query_bbox_d =
+    on_device ? BBoxArray(query_bbox_h, kernel_allocator) : BBoxArray();
+  auto query_bbox_v = on_device ? query_bbox_d.view() : query_bbox_h.view();
+
+  axom::utilities::Timer timer;
+
+  // Initialize a BVH tree over the insert mesh bounding boxes
+  timer.start();
+  axom::spin::BVH<3, ExecSpace, double> bvh;
+  bvh.setAllocatorID(kernel_allocator);
+  bvh.initialize(insert_bbox_v, insert_bbox_v.size());
+  timer.stop();
+  SLIC_INFO(axom::fmt::format("0: Initializing BVH took {:4.3} seconds.",
+                              timer.elapsedTimeInSec()));
+
+  // Search for candidate bounding boxes of hexes to query;
+  timer.start();
+  IndexArray offsets_d(query_bbox_v.size(), query_bbox_v.size(), kernel_allocator);
+  IndexArray counts_d(query_bbox_v.size(), query_bbox_v.size(), kernel_allocator);
+  IndexArray candidates_d(0, 0, kernel_allocator);
+
+  auto offsets_v = offsets_d.view();
+  auto counts_v = counts_d.view();
+  bvh.findBoundingBoxes(offsets_v,
+                        counts_v,
+                        candidates_d,
+                        query_bbox_v.size(),
+                        query_bbox_v);
+
+  timer.stop();
+  SLIC_INFO(axom::fmt::format(
+    "1: Querying candidate bounding boxes took {:4.3} seconds.",
+    timer.elapsedTimeInSec()));
+
+  // Initialize candidatePairs to return
+  timer.start();
+
+  IndexArray offsets_h(offsets_d, host_allocator);
+  IndexArray counts_h(counts_d, host_allocator);
+  IndexArray candidates_h(candidates_d, host_allocator);
+
+  for(int i = 0; i < queryMesh.numHexes(); i++)
+  {
+    for(int j = 0; j < counts_h[i]; j++)
+    {
+      candidatePairs.emplace_back(
+        std::make_pair(i, candidates_h[offsets_h[i] + j]));
+    }
+  }
+  timer.stop();
```

Comment:
Can this expansion be RAJA-fied (here and for the ImplicitGrid) ?

In addition to performing this using the execution policy, it would avoid the memory operations.
Presumably, any follow-up computational geometry would need to be done on device.